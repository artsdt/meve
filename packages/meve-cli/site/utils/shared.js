import Vue from 'vue'

export const isArray = val => Array.isArray(val)

export function getMenuName(path) {
  return path.slice(path.lastIndexOf('/') + 1)
}

export function getLang(path) {
  return path.slice(1, path.lastIndexOf('/'))
}

export function buildTreeRelation(tree, parent, seen) {
  const flatTree = []

  tree.forEach((node) => {
    seen ? seen.push(node) : flatTree.push(node)
    node.parent = parent
    node.leaf = false

    if (isArray(node.children)) {
      buildTreeRelation(node.children, node, flatTree)
    } else {
      node.leaf = true
    }
  })

  return {
    tree,
    flatTree
  }
}

export function findNodeRelationChain(node) {
  const chain = []
  let parent = node.parent

  while (parent) {
    chain.unshift(parent)
    parent = parent.parent
  }

  return chain
}

export function nextTick() {
  return new Promise((resolve) => Vue.nextTick(resolve))
}

export function doubleRaf() {
  return new Promise((resolve) => {
    window.requestAnimationFrame(() => {
      window.requestAnimationFrame(resolve)
    })
  })
}

