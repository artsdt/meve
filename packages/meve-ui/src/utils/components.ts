import Vue, { ComponentOptions } from 'vue'
import { CombinedVueInstance } from 'vue/types/vue'

export interface MountComponentApi {
  instance: CombinedVueInstance<any, any, any, any, any>;
  unmount(): void;
}

export function mountComponent(component: ComponentOptions<Vue>, container: string, options = {}): MountComponentApi {
  const instance = new (Vue.extend(component))(options)
  const el = instance.$mount().$el
  const wrapper = document.querySelector(container)!
  wrapper.appendChild(el)

  return {
    instance,
    unmount() {
      instance.$destroy()
      wrapper.removeChild(el)
    },
  }
}

export function nextTick(): Promise<void> {
  return new Promise((resolve) => Vue.nextTick(resolve))
}
