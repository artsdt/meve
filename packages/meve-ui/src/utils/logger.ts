export function warn(message: string, from?: string) {
  if (process?.env?.NODE_ENV === 'development') {
    console.warn(`Warning: ${message} ${from ? `from ${from}` : ''}`)
  }
}
