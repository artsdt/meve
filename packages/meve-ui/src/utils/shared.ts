export interface CacheInstance<T> {
  cache: T[];

  has(key: T): boolean;

  add(key: T): void;

  remove(key: T): void;

  clear(): void;
}

export const isString = (val: unknown): val is string => typeof val === 'string'

export const toNumber = (val: number | string | boolean | undefined | null): number => {
  if (val == null) return 0

  if (isString(val)) {
    val = parseFloat(val)
    val = Number.isNaN(val) ? 0 : val
    return val
  }

  if (isBoolean(val)) return Number(val)

  return val
}

export const isBoolean = (val: unknown): val is boolean => typeof val === 'boolean'

export const isFunction = (val: unknown): val is Function => typeof val === 'function'

export const isNumber = (val: unknown): val is number => typeof val === 'number'

export const isPlainObject = (val: unknown): val is Record<string, any> =>
  Object.prototype.toString.call(val) === '[object Object]'

export const isObject = (val: unknown): val is object => typeof val === 'object' && val !== null

export const isArray = (val: unknown): val is Array<any> => Array.isArray(val)

export const isEmpty = (val: unknown) =>
  val === undefined || val === null || val === '' || (Array.isArray(val) && !val.length)

export const removeItem = (arr: Array<unknown>, item: unknown) => {
  if (arr.length) {
    const index: number = arr.indexOf(item)
    if (index > -1) {
      return arr.splice(index, 1)
    }
  }
}

export const toggleItem = (arr: Array<unknown>, item: unknown) => {
  arr.includes(item) ? removeItem(arr, item) : arr.push(item)
}

export const throttle = (method: any, mustRunDelay = 200): (() => void) => {
  let timer: number
  let start = 0

  return function loop(this: unknown, ...args) {
    const now = Date.now()
    const elapsed = now - start

    if (!start) {
      start = now
    }

    if (timer) {
      window.clearTimeout(timer)
    }

    if (elapsed >= mustRunDelay) {
      method.apply(this, args)
      start = now
    } else {
      timer = window.setTimeout(() => {
        loop.apply(this, args)
      }, mustRunDelay - elapsed)
    }
  }
}

export const debounce = (method: any, delay = 200) => {
  let timer: number

  return function (this: unknown, ...args: any[]) {
    timer && window.clearTimeout(timer)
    timer = window.setTimeout(() => {
      method.apply(this, args)
    }, delay)
  }
}

export function kebabCase(str: string): string {
  const reg = /([^-])([A-Z])/g

  return str.replace(reg, '$1-$2').replace(reg, '$1-$2').toLowerCase()
}

export const camelize = (s: string): string => s.replace(/-(\w)/g, (_: any, p: string) => p.toUpperCase())

export const bigCamelize = (s: string): string => camelize(s).replace(s.charAt(0), s.charAt(0).toUpperCase())

export const range = (num: number, left: number, right: number) => {
  if (num < left) {
    return left
  }

  if (num > right) {
    return right
  }

  return num
}

export const delay = (time: number) => new Promise((resolve) => setTimeout(resolve, time))

export const inBrowser = () => typeof window !== 'undefined'

export const createCache = <T>(max: number): CacheInstance<T> => {
  const cache: T[] = []

  return {
    cache,

    has(key: T) {
      return this.cache.includes(key)
    },

    add(key: T) {
      if (this.has(key)) {
        return
      }

      this.cache.length === max && cache.shift()
      this.cache.push(key)
    },

    remove(key: T) {
      this.has(key) && removeItem(this.cache, key)
    },

    clear() {
      this.cache.length = 0
    },
  }
}

export const uniq = (arr: Array<any>) => [...new Set(arr)]

export const NOOP = () => {}
