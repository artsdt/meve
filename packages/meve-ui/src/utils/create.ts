import type Vue from 'vue'
import { VNodeData, ComponentOptions, VueConstructor, CreateElement, VNode, RenderContext } from 'vue'
import { PropsDefinition } from 'vue/types/options'
import { bigCamelize, isArray, isFunction, kebabCase } from './shared'
import { SlotsMixin } from './mixins/slots'

export type DefaultProps = Record<string, any>

export type DefaultSlots = {
  default?: ScopedSlot;
}

export type ScopedSlot<Props = any> = (props?: Props) => VNode[] | undefined

export type ScopedSlots = {
  [key: string]: ScopedSlot | undefined;
} & DefaultSlots

export type FunctionComponent<Props = DefaultProps, PropsDefinitions = PropsDefinition<Props>> = {
  (h: CreateElement, props: Props, slots: ScopedSlots, context: RenderContext<Props>): VNode | undefined;
  name?: string;
  props?: PropsDefinitions;
}

export type FunctionComponentPlugin = FunctionComponent & ComponentPlugin

export type TsxBaseProps<Slots> = {
  key: string | number;
  props: any;
  class: any;
  style: string | object[] | object;
  scopedSlots: Slots;
}

export type TsxComponent<Props, Events, Slots> = (props: Partial<Props & Events & TsxBaseProps<Slots>>) => VNode

export interface ComponentPlugin {
  install?: (app: VueConstructor, options?: any) => void;
}

export type ComponentOptionsPlugin = ComponentOptions<Vue> & ComponentPlugin

export function mergeSlots(context: RenderContext) {
  const scopedSlots = context.scopedSlots ?? context.data.scopedSlots ?? {}
  const slots = context.slots()

  Object.keys(slots).forEach((key) => {
    if (!scopedSlots[key]) {
      scopedSlots[key] = () => slots[key]
    }
  })

  return scopedSlots
}

export function wrapFunctionComponent(
  functionComponent: FunctionComponent
): ComponentOptionsPlugin & { functional?: boolean } {
  return {
    functional: true,
    props: functionComponent.props,
    render: (h: CreateElement, context: RenderContext): any =>
      functionComponent(h, context.props, mergeSlots(context), context),
  }
}

export function install(this: ComponentOptions<Vue>, Vue: VueConstructor) {
  Vue.component(this.name as string, this)
}

export function createComponent<Props = DefaultProps, Events = {}, Slots = {}>(
  name: string,
  sfc: ComponentOptionsPlugin | FunctionComponentPlugin
) {
  if (isFunction(sfc)) {
    sfc = wrapFunctionComponent(sfc as FunctionComponentPlugin)
  } else if ((sfc as ComponentOptionsPlugin).mixins) {
    ;(sfc as ComponentOptionsPlugin).mixins!.push(SlotsMixin)
  } else {
    ;(sfc as ComponentOptionsPlugin).mixins = [SlotsMixin]
  }

  sfc.name = `M${bigCamelize(name)}`
  sfc.install = install

  return sfc as TsxComponent<Props, Events, Slots>
}

export function createNamespace(name: string) {
  return {
    createComponent<Props, Events, Slots>(sfc: ComponentOptionsPlugin | FunctionComponentPlugin) {
      return createComponent<Props, Events, Slots>(name, sfc)
    },
    namespace: (suffix = '') => `m-${kebabCase(name)}${suffix}`,
  }
}

export type Context = RenderContext & { data: VNodeData & Record<string, any> }

export type InheritContext = Partial<VNodeData> & Record<string, any>

const inheritKeys = [
  'ref',
  'key',
  'style',
  'class',
  'attrs',
  'refInFor',
  'nativeOn',
  'directives',
  'staticClass',
  'staticStyle',
]

export function inherit(context: Context): InheritContext {
  return inheritKeys.reduce((ret, key) => {
    if (context.data[key]) {
      ret[key === 'nativeOn' ? 'on' : key] = context.data[key]
    }
    return ret
  }, {} as InheritContext)
}

export function emit(context: RenderContext, eventName: string, ...args: any[]) {
  const listeners = context.listeners[eventName]
  if (isArray(listeners)) {
    listeners.forEach((listener) => listener(...args))
  } else {
    listeners?.(...args)
  }
}
