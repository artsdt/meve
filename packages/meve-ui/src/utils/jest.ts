import Vue from 'vue'
import { Wrapper, config } from '@vue/test-utils'
import { isPlainObject } from './shared'

export const delay = (time = 0) => new Promise((resolve) => setTimeout(resolve, time))

export function mockOffset({
  offsetWidth,
  offsetHeight,
  clientWidth,
  clientHeight,
  offsetLeft,
  offsetTop,
  scrollWidth,
  scrollHeight,
}: {
  offsetWidth?: number;
  offsetHeight?: number;
  clientWidth?: number;
  clientHeight?: number;
  offsetLeft?: number;
  offsetTop?: number;
  scrollWidth?: number;
  scrollHeight?: number;
} = {}) {
  Object.defineProperties(HTMLElement.prototype, {
    offsetWidth: {
      get() {
        return offsetWidth ?? (parseFloat(window.getComputedStyle(this).width) || 0)
      },
    },
    offsetHeight: {
      get() {
        return offsetHeight ?? (parseFloat(window.getComputedStyle(this).height) || 0)
      },
    },
    clientWidth: {
      get() {
        return clientWidth ?? (parseFloat(window.getComputedStyle(this).width) || 0)
      },
    },
    clientHeight: {
      get() {
        return clientHeight ?? (parseFloat(window.getComputedStyle(this).height) || 0)
      },
    },
    offsetLeft: {
      get() {
        return offsetLeft ?? (parseFloat(window.getComputedStyle(this).marginLeft) || 0)
      },
    },
    offsetTop: {
      get() {
        return offsetTop ?? (parseFloat(window.getComputedStyle(this).marginTop) || 0)
      },
    },
    offsetParent: {
      get() {
        return this.parentNode ?? {}
      },
    },
    scrollWidth: {
      get() {
        return scrollWidth ?? 0
      },
    },
    scrollHeight: {
      get() {
        return scrollHeight ?? 0
      },
    },
  })

  return {
    mockRestore() {
      Object.defineProperties(HTMLElement.prototype, {
        offsetWidth: {
          get() {
            return parseFloat(window.getComputedStyle(this).width) || 0
          },
        },
        offsetHeight: {
          get() {
            return parseFloat(window.getComputedStyle(this).height) || 0
          },
        },
        clientWidth: {
          get() {
            return parseFloat(window.getComputedStyle(this).width) || 0
          },
        },
        clientHeight: {
          get() {
            return parseFloat(window.getComputedStyle(this).height) || 0
          },
        },
        offsetLeft: {
          get() {
            return parseFloat(window.getComputedStyle(this).marginLeft) || 0
          },
        },
        offsetTop: {
          get() {
            return parseFloat(window.getComputedStyle(this).marginTop) || 0
          },
        },
        offsetParent: {
          get() {
            return this.parentNode ?? {}
          },
        },
        scrollWidth: {
          get() {
            return 0
          },
        },
        scrollHeight: {
          get() {
            return 0
          },
        },
      })
    },
  }
}

export async function triggerDrag(
  el: Wrapper<Vue> | HTMLElement | Window,
  x = 0,
  y = 0,
  triggerTouchend = true,
  triggerDocument = false
) {
  const delegationEl = triggerDocument ? document : el

  await trigger(el, 'touchstart', 0, 0)
  await trigger(delegationEl, 'touchmove', x / 4, y / 4)
  await trigger(delegationEl, 'touchmove', x / 3, y / 3)
  await trigger(delegationEl, 'touchmove', x / 2, y / 2)
  await trigger(delegationEl, 'touchmove', x, y)

  if (triggerTouchend) {
    await trigger(delegationEl, 'touchend', x, y)
  }
}

export function getTouch(el: Element | Window | Document, x: number, y: number) {
  return {
    identifier: Date.now(),
    target: el,
    pageX: x,
    pageY: y,
    clientX: x,
    clientY: y,
    radiusX: 2.5,
    radiusY: 2.5,
    rotationAngle: 10,
    force: 0.5,
  }
}

export function trigger(
  wrapper: Wrapper<Vue> | HTMLElement | Window | Document,
  eventName: string,
  x = 0,
  y = 0,
  deltaX = 0,
  deltaY = 0,
  offsetX = 0,
  offsetY = 0,
  args = {}
) {
  const el = 'element' in wrapper ? wrapper.element : wrapper
  const touchList = [getTouch(el, x, y)]

  const event = document.createEvent('CustomEvent')
  event.initCustomEvent(eventName, true, true, {})

  Object.assign(event, {
    clientX: x,
    clientY: y,
    deltaX,
    deltaY,
    offsetX,
    offsetY,
    touches: touchList,
    targetTouches: touchList,
    changedTouches: touchList,
    ...args,
  })

  el.dispatchEvent(event)

  return Vue.nextTick()
}

export function mockStubs() {
  const originStubs = config.stubs

  config.stubs = {}

  return {
    mockRestore() {
      config.stubs = originStubs
    },
  }
}

export function mockDoubleRaf() {
  const originMethod = window.requestAnimationFrame

  Object.assign(window, {
    requestAnimationFrame(fn: any) {
      setTimeout(fn, 16)
    },
  })

  return {
    mockRestore() {
      window.requestAnimationFrame = originMethod
    },
  }
}

export function mockScrollTo(Element: any) {
  Element.prototype.scrollTo = function (x: ScrollToOptions | number, y?: number) {
    if (isPlainObject(x)) {
      this.scrollLeft = x.left
      this.scrollTop = x.top
    } else {
      this.scrollLeft = x
      this.scrollTop = y
    }
  }
}
