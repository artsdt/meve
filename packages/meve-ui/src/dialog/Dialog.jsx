import Popup from '../popup'
import Button from '../button'
import Icon from '../icon'
import { createNamespace } from '../utils/create'
import { props } from './props'

import '../button/button.less'
import '../icon/icon.less'
import '../popup/popup.less'
import '../styles/common.less'
import './dialog.less'

const { createComponent, namespace } = createNamespace('dialog')

const DialogPlugin = createComponent({
  props,
  data: () => ({
    show: false,
    localCloseOnClickOverlay: false,
  }),

  watch: {
    value: {
      handler(newValue) {
        this.show = newValue
      },
      immediate: true,
    },

    closeOnClickOverlay: {
      handler(newValue) {
        if (this.beforeClose) {
          this.localCloseOnClickOverlay = false
          return
        }

        this.localCloseOnClickOverlay = newValue
      },
      immediate: true,
    },
  },

  methods: {
    handleOpen() {
      this.$emit('open')
    },

    handleOpened() {
      this.$emit('opened')
    },

    handleClose() {
      this.$emit('close')
    },

    handleClosed() {
      this.$emit('closed')
    },

    handleRouteChange() {
      this.$emit('route-change')
    },

    done() {
      this.$emit('input', false)
    },

    handleClickOverlay() {
      this.$emit('click-overlay')

      if (!this.closeOnClickOverlay) {
        return
      }

      if (this.beforeClose) {
        this.beforeClose('close', this.done.bind(this))
        return
      }

      this.done()
    },

    handleClickCloseButton() {
      if (this.beforeClose) {
        this.beforeClose('close', this.done.bind(this))
        return
      }

      this.done()
    },

    confirm() {
      this.$emit('confirm')

      if (this.beforeClose) {
        this.beforeClose('confirm', this.done.bind(this))
        return
      }

      this.done()
    },

    cancel() {
      this.$emit('cancel')

      if (this.beforeClose) {
        this.beforeClose('cancel', this.done.bind(this))
        return
      }

      this.done()
    },

    renderMessage() {
      if (this.hasSlots()) {
        return this.slots()
      }

      if (this.message) {
        return (
          <div class={namespace('__message')} style={{ textAlign: this.messageAlign }}>
            {this.message}
          </div>
        )
      }
    },

    renderTitle() {
      if (this.hasSlots('title')) {
        return this.slots('title')
      }

      return <div class={namespace('__title')}>{this.title ?? '提示'}</div>
    },

    renderFooter() {
      if (this.hasSlots('footer')) {
        return this.slots('footer')
      }

      return (
        <div class={namespace('__footer')}>
          {this.confirmButton && (
            <Button
              class={namespace('__confirm-button')}
              dialog-cover
              size="large"
              type={this.confirmButtonType}
              onClick={this.confirm}
            >
              {this.confirmButtonText ?? '确认'}
            </Button>
          )}
          {this.cancelButton && (
            <Button
              class={namespace('__cancel-button')}
              dialog-cover
              size="large"
              type={this.cancelButtonType}
              text={true}
              onClick={this.cancel}
            >
              {this.cancelButtonText ?? '取消'}
            </Button>
          )}
        </div>
      )
    },
  },

  render() {
    const {
      show,
      overlay,
      overlayClass,
      lockScroll,
      localCloseOnClickOverlay,
      handleOpen,
      handleOpened,
      handleClose,
      handleClosed,
      handleClickOverlay,
      handleRouteChange,
    } = this

    return (
      <Popup
        class={namespace()}
        dialog-cover
        value={show}
        overlay={overlay}
        overlayClass={overlayClass}
        lockScroll={lockScroll}
        closeOnClickOverlay={localCloseOnClickOverlay}
        onOpen={handleOpen}
        onOpened={handleOpened}
        onClose={handleClose}
        onClosed={handleClosed}
        onClickOverlay={handleClickOverlay}
        onRouteChange={handleRouteChange}
      >
        <div class={namespace('__container')}>
          <div class={namespace('__title-wrapper')}>
            {this.renderTitle()}
            <Button
              class={namespace('__close-button')}
              dialog-cover
              text={true}
              round={true}
              onClick={this.handleClickCloseButton}
            >
              <Icon name="close" />
            </Button>
          </div>

          {this.renderMessage()}
          {this.renderFooter()}
        </div>
      </Popup>
    )
  },
})

export const _DialogComponent = DialogPlugin

export default DialogPlugin
