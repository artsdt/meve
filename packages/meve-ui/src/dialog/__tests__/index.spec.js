import Dialog from '..'
import Vue from 'vue'
import { delay, trigger } from '../../utils/jest'

test('test dialog function plugin', () => {
  Vue.use(Dialog)
  expect(Vue.component(Dialog.Component.name)).toBeTruthy()
})

test('test dialog function pass string', async () => {
  Dialog('hello world')
  await delay()
  expect(document.body.innerHTML).toMatchSnapshot()
  Dialog.close()
  await delay(300)
  expect(document.body.innerHTML).toBeFalsy()
})

test('test dialog function pass events callback', async () => {
  const onConfirm = jest.fn()
  const onCancel = jest.fn()
  const onClose = jest.fn()
  const onClosed = jest.fn()
  const onOpen = jest.fn()
  const onOpened = jest.fn()
  const onBeforeClose = jest.fn()
  const onClickOverlay = jest.fn()

  Dialog({
    title: 'this is title',
    message: 'this is message',
    onConfirm,
    onCancel,
    onClose,
    onClosed,
    onOpen,
    onOpened,
    onBeforeClose,
    onClickOverlay,
  })

  await delay()
  expect(onOpen).toHaveBeenCalledTimes(1)
  await delay(300)
  expect(onOpened).toHaveBeenCalledTimes(1)

  await trigger(document.querySelector('.m-dialog__confirm-button'), 'click')
  expect(onConfirm).toHaveBeenCalledTimes(1)

  await trigger(document.querySelector('.m-dialog__cancel-button'), 'click')
  expect(onCancel).toHaveBeenCalledTimes(1)

  await trigger(document.querySelector('.m-popup__overlay'), 'click')
  expect(onClickOverlay).toHaveBeenCalledTimes(1)

  await trigger(document.querySelector('.m-dialog__close-button'), 'click')

  expect(onBeforeClose).toHaveBeenCalledTimes(4)

  Dialog.close()
  await delay()

  expect(onClose).toHaveBeenCalledTimes(1)
  await delay(300)
  expect(onClosed).toHaveBeenCalledTimes(1)
})
