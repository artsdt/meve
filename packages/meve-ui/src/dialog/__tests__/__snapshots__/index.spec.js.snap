// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test dialog function pass string 1`] = `"<div class=\\"m--box m-popup m-dialog m-fade-enter m-fade-enter-active\\" style=\\"z-index: 1999;\\"><div class=\\"m-popup__overlay\\" style=\\"z-index: 2000;\\"></div><div dialog-cover=\\"true\\" class=\\"m-popup__content m-popup--center m-pop-center-enter m-pop-center-enter-active\\" style=\\"z-index: 2001;\\"><div class=\\"m-dialog__container\\"><div class=\\"m-dialog__title-wrapper\\"><div class=\\"m-dialog__title\\">提示</div><button tabindex=\\"-1\\" class=\\"m-button m--box m-button--normal m--inline-flex m-button--text-default m-button--text m-button--round m-dialog__close-button\\" dialog-cover=\\"true\\"><div class=\\"m-button__content\\"><i class=\\"m-icon m--box meve-icon--set meve-icon-close\\"></i></div></button></div><div class=\\"m-dialog__message\\" style=\\"text-align: left;\\">hello world</div><div class=\\"m-dialog__footer\\"><button tabindex=\\"-1\\" class=\\"m-button m--box m-button--large m--inline-flex m-button--primary m-button--shadow m-dialog__confirm-button\\" dialog-cover=\\"true\\"><div class=\\"m-button__content\\">确认</div></button><button tabindex=\\"-1\\" class=\\"m-button m--box m-button--large m--inline-flex m-button--text-primary m-button--text m-dialog__cancel-button\\" dialog-cover=\\"true\\"><div class=\\"m-button__content\\">取消</div></button></div></div></div></div>"`;
